builder/dockerfile/builder.go

```diff
@@ -196,6 +199,11 @@ func (b *Builder) dispatchDockerfileWithCancellation(dockerfile *parser.Result)
 		// TODO: get this from dispatch
 		imageID = b.image
 
+		if b.Aux != nil {
+			if err := b.Aux.Emit(types.BuildResult{ID: imageID}); err != nil {
+				return "", err
+			}
+		}
```

Comment:
I meant ID for each stage but I guess we could also leave it out for now(wdyt @dnephin ) and create a separate issue for this problem(in cli you can leave it so it only saves last one). If we do report stages then best place currently would be here but you need to check for the stage switch with a from command(like line 188).